pkg/controllers/management/auth/globalroles/globalrolebinding_handler_test.go

```diff
@@ -1146,3 +1175,590 @@ func Test_reconcileClusterPermissions(t *testing.T) {
+					return nil, fmt.Errorf("error")
+				}
+				state.rbListerMock.ListFunc = func(_ string, _ labels.Selector) ([]*rbacv1.RoleBinding, error) {
+					return nil, nil
+				}
+			},
+			globalRoleBinding: namespacedRulesGRB.DeepCopy(),
+			wantError:         true,
+		},
+		{
+			name: "roleBindings get created",
+			stateSetup: func(state grbTestState) {
+				state.grListerMock.GetFunc = func(_, _ string) (*v3.GlobalRole, error) {
+					return namespacedRulesGR.DeepCopy(), nil
+				}
+				state.nsCacheMock.EXPECT().Get(gomock.Any()).AnyTimes().Return(activeNamespace, nil)
+
+				state.rListerMock.GetFunc = func(_, _ string) (*rbacv1.Role, error) {
+					return nil, nil
+				}
+				state.rbListerMock.GetFunc = func(_, _ string) (*rbacv1.RoleBinding, error) {
+					return nil, apierrors.NewNotFound(schema.GroupResource{
+						Group:    normanv1.RoleBindingGroupVersionKind.Group,
+						Resource: normanv1.RoleBindingGroupVersionResource.Resource,
+					}, "")
+				}
+				state.rbClientMock.CreateFunc = func(roleBinding *rbacv1.RoleBinding) (*rbacv1.RoleBinding, error) {
+					state.stateChanges.createdRBs[roleBinding.Name] = roleBinding
+					roleBinding.UID = ""
+					return roleBinding, nil
+				}
+				state.rbListerMock.ListFunc = func(_ string, _ labels.Selector) ([]*rbacv1.RoleBinding, error) {
+					return nil, nil
+				}
+			},
+			stateAssertions: func(stateChanges grbTestStateChanges) {
+				require.Len(stateChanges.t, stateChanges.createdRBs, 2)
+
+				roleBinding, ok := stateChanges.createdRBs["namespacedRulesGRB-namespace1"]
+				require.True(stateChanges.t, ok)
+				require.Equal(stateChanges.t, "namespace1", roleBinding.ObjectMeta.Namespace)
+				require.Equal(stateChanges.t, "namespacedRulesGR-namespace1", roleBinding.RoleRef.Name)
+				require.Equal(stateChanges.t, "Role", roleBinding.RoleRef.Kind)
+				require.Equal(stateChanges.t, "namespacedRulesGRB", roleBinding.Labels[grbOwnerLabel])
+				subject := rbacv1.Subject{
+					Kind:     "User",
+					Name:     "username",
+					APIGroup: rbacv1.GroupName,
+				}
+				require.Len(stateChanges.t, roleBinding.Subjects, 1)
+				require.Equal(stateChanges.t, subject, roleBinding.Subjects[0])
+
+				roleBinding, ok = stateChanges.createdRBs["namespacedRulesGRB-namespace2"]
+				require.True(stateChanges.t, ok)
+				require.Equal(stateChanges.t, "namespace2", roleBinding.ObjectMeta.Namespace)
+				require.Equal(stateChanges.t, "namespacedRulesGR-namespace2", roleBinding.RoleRef.Name)
+				require.Equal(stateChanges.t, "Role", roleBinding.RoleRef.Kind)
+				require.Len(stateChanges.t, roleBinding.Subjects, 1)
+				require.Equal(stateChanges.t, subject, roleBinding.Subjects[0])
+				require.Equal(stateChanges.t, "namespacedRulesGRB", roleBinding.Labels[grbOwnerLabel])
+			},
+			globalRoleBinding: namespacedRulesGRB.DeepCopy(),
+			wantError:         false,
+		},
+		{
+			name: "roleBindings get created in terminating namespace",
```

Comment:
Nit: the name seems a bit innacurate, since we don't want the bindings to be made in a terminating namespace.

Reply:
Clarified in the name